impl_core/src/main/java/io/opencensus/implcore/trace/export/SampledSpanStoreImpl.java

```diff
@@ -252,7 +256,11 @@ public Summary getSummary() {
    */
   public void considerForSampling(SpanImpl span) {
     synchronized (samples) {
-      PerSpanNameSamples perSpanNameSamples = samples.get(span.getName());
+      String spanName = span.getName();
+      if (span.getSampleToLocalSpanStore() && !samples.containsKey(spanName)) {
```

Comment:
You can get rid of the synchronization in this class by changing `samples` to a `ConcurrentHashMap`.
Here and in `registerSpanNamesForCollection()` you can use `putIfAbsent()`:
```java
PerSpanNameSamples perSpanNameSamples = samples.get(spanName);
if (perSpanNameSamples == null && span.getSampleToLocalSpanStore()) {
  perSpanNameSamples = new PerSpanNameSamples();
  PerSpanNameSamples existingSamples = samples.putIfAbsent(spanName, perSpanNameSamples);
  if (existingSamples != null) {
    perSpanNameSamples = existingSamples;
  }
}
```

Reply:
https://github.com/census-instrumentation/opencensus-java/issues/711